strongbox-web-core/src/main/java/org/carlspring/strongbox/interceptors/RepositoryRequestInterceptor.java

```diff
@@ -0,0 +1,100 @@
+package org.carlspring.strongbox.interceptors;
+
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class RepositoryRequestInterceptor
+        extends HandlerInterceptorAdapter
+{
+
+    private final String storageIdPathVariableName;
+
+    private final String repositoryIdPathVariableName;
+
+    private final String exposedRepositoryRequestAttributeName;
+
+    private final ConfigurationManagementService configurationManagementService;
+
+    public RepositoryRequestInterceptor(String storageIdPathVariableName,
+                                        String repositoryIdPathVariableName,
+                                        String exposedRepositoryRequestAttributeName,
```

Comment:
@fuss86 what is the reason not to use constants directly instead of this parameters?

Reply:
You can use this interceptor for other PathVariables like `srcStorageId`/`srcRepositoryId` for example

Reply:
ok, but this wouldn't be required in case of [last](https://github.com/strongbox/strongbox/pull/1272#discussion_r287205740) proposed solution

Reply:
but it still provides more flexibility for future cases like the one I noted above

Reply:
nevermind, done :)